torch/_inductor/fx_passes/replace_random.py

```diff
@@ -0,0 +1,194 @@
+import collections
+import functools
+import logging
+
+import torch
+from ..._dynamo.utils import counters
+from .. import config, inductor_prims
+from ..pattern_matcher import (
+    CallFunctionVarArgs,
+    inference_graph,
+    init_once_fakemode,
+    Match,
+    PatternMatcherPass,
+    register_graph_pattern,
+    register_replacement,
+    training_graph,
+)
+from ..virtualized import V
+
+log = logging.getLogger(__name__)
+patterns = PatternMatcherPass()
+aten = torch.ops.aten
+
+
+def replace_random_passes(gm: torch.fx.GraphModule):
+    """Modify the given FX graph to use backend-native random ops"""
+    if config.fallback_random:
+        return 0
+
+    lazy_init()
+
+    count = patterns.apply(gm)
+    count += fuse_seed_creation_pass(gm.graph)
+
+    return count
+
+
+def fuse_seed_creation_pass(graph: torch.fx.Graph):
```

Comment:
but we usually still have 1 seed per graph?

Reply:
This changes things so each call to rand() uses a different seed, all generated with a single call to aten.randint.  This removes needing to accumulate offsets across kernels, which is: 1) messy with dynamic shapes; 2) suffers from int32 overflow issues

Reply:
General recommendations for high quality prng sequences are to use generated sequence for a single seed because sequences from different seeds can be correlated. Dynamic shapes worked, on the examples I looked at at least, wrapping around for overflown offsets that we were doing also introduced hard limit to the sequence length, but I don't know if it's better/worse than starting from the different seeds every time.